src/mesh/high_order_grid.cpp

```diff
@@ -1050,6 +1050,24 @@ void HighOrderGrid<dim,real,MeshType,VectorType,DoFHandlerType>::execute_coarsen
     //}
 }
 
+template <int dim, typename real, typename MeshType, typename VectorType, typename DoFHandlerType>
+dealii::Point<dim> HighOrderGrid<dim,real,MeshType,VectorType,DoFHandlerType>::smallest_cell_coordinates()
+{
+    dealii::Point<dim> smallest_cell_coord;
+    double min_diameter = triangulation->begin_active()->diameter();
+    for (const auto &cell : triangulation->active_cell_iterators()) 
+    {
+        if (min_diameter > cell->diameter())
+        {
+            min_diameter = cell->diameter();
+            smallest_cell_coord = cell->vertex(0);
```

Comment:
Any reason you're just querying the first cell's vertex as the cell's location?

Reply:
Nevermind, I now understand that it's just meant for the test.

Reply:
Yeah, it's just to get a ball park estimate of the refinement location.